account_check_printing_report_base/report/check_print.py

```diff
@@ -92,16 +94,15 @@ def get_paid_lines(self, payments):
 
     @api.multi
     def get_report_values(self, docids, data=None):
-        payments = self.env['account.payment'].browse(docids)
-        paid_lines = self.get_paid_lines(payments)
+        model = self.env.context.get('active_model', 'account.payment')
+        objects = self.env[model].browse(docids)
+        paid_lines = self.get_paid_lines(objects)
         docargs = {
             'doc_ids': docids,
-            'doc_model': 'account.payment',
-            'docs': payments,
+            'doc_model': models,
+            'docs': objects,
             'time': time,
             'fill_stars': self.fill_stars,
             'paid_lines': paid_lines
         }
-        if self.env.user.company_id.check_layout_id:
```

Comment:
I think this is still needed

Reply:
If the layout is not defined, the following error wil be raised: https://github.com/odoo/odoo/blob/11.0/addons/account_check_printing/models/account_payment.py#L120

Here, you are already on a report, so I think it is not necessary to check the layout.

Reply:
Is that method called always?

Reply:
Not necessarly, but If you have selected a layout in order to print a check, why should it fail if your company (not the company of the payment) have no layout for the checks defined?

Reply:
But how do you select that layout? For now, there's no other option.

Reply:
Yes, but the layout is selected earlier on https://github.com/OCA/account-payment/blob/11.0/account_check_printing_report_base/models/account_payment.py#L30. If no layout is selected, it will raise the error I showed.

Reply:
OK, but next time please don't make these extra changes without advertising, as this is not the aim of this PR.